File: apps/backend/src/EnterpriseAgentOs.Domain/Common/ValueObjects/OAuthProvider.cs
namespace EnterpriseAgentOs.Domain.Common.ValueObjects;

public enum OAuthProvider
{
    Google,
    Microsoft,
    GitHub,
}

public static class OAuthProviderExtensions
{
    public static string ToStorageString(this OAuthProvider provider) => provider switch
    {
        OAuthProvider.Google => "google",
        OAuthProvider.Microsoft => "microsoft",
        OAuthProvider.GitHub => "github",
        _ => throw new ArgumentOutOfRangeException(nameof(provider)),
    };

    public static OAuthProvider ToOAuthProvider(this string value) => value switch
    {
        "google" => OAuthProvider.Google,
        "microsoft" => OAuthProvider.Microsoft,
        "github" => OAuthProvider.GitHub,
        _ => throw new ArgumentOutOfRangeException(nameof(value), $"Unknown OAuth provider: {value}"),
    };
}

File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/common/valueobjects/OAuthProvider.java
package com.enterpriseagentos.backendjava.domain.common.valueobjects;

import java.util.Locale;

public enum OAuthProvider {
    Google,
    Microsoft,
    GitHub;

    public String toWire() {
        return switch (this) {
            case Google -> "google";
            case Microsoft -> "microsoft";
            case GitHub -> "github";
        };
    }

    public static OAuthProvider fromWire(String value) {
        return switch (value.trim().toLowerCase(Locale.ROOT)) {
            case "google" -> Google;
            case "microsoft" -> Microsoft;
            case "github" -> GitHub;
            default -> throw new IllegalArgumentException("Unknown OAuth provider: " + value);
        };
    }
}
